gobblin-metrics-libs/gobblin-metrics-base/src/main/java/org/apache/gobblin/metrics/ContextAwareMetricFactory.java

```diff
@@ -108,8 +129,8 @@ public ContextAwareHistogram newMetric(MetricContext context, String name) {
     }
 
     @Override
-    public ContextAwareHistogram newMetric(MetricContext context, String name, Object ... args) {
-      return new ContextAwareHistogram(context, name, (long)args[0], (TimeUnit) args[1]);
+    public ContextAwareHistogram newMetric(ContextAwareMetricFactoryArgs args) {
```

Comment:
The `ContextAwareMetricFactoryArgs` should be unknown to object. Only factory knows the factory args and decides which constructor to invoke depending on the specific type of `ContextAwareMetricFactoryArgs`